src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs

```diff
@@ -679,19 +724,28 @@ public interface OnFooListener
 		[Test]
 		[Category ("AOT")]
 		[NonParallelizable]
-		public void BuildApplicationWithSpacesInPath ([Values (true, false)] bool enableMultiDex, [Values ("", "r8")] string linkTool)
+		public void BuildApplicationWithSpacesInPath ([Values (true, false)] bool enableMultiDex, [Values ("", "r8")] string linkTool, [Values] AndroidRuntime runtime)
 		{
-			var folderName = $"BuildReleaseApp AndÜmläüts({enableMultiDex}{linkTool})";
+			const bool isRelease = true;
+			bool aotAssemblies = runtime == AndroidRuntime.MonoVM;
+			if (IgnoreUnsupportedConfiguration (runtime, aot: aotAssemblies, release: isRelease)) {
+				return;
+			}
+
+			var folderName = Path.Combine (Root, "temp", TestName);
 			var lib = new XamarinAndroidLibraryProject {
 				IsRelease = true,
 				ProjectName = "Library1"
 			};
+			lib.SetRuntime (runtime);
+
 			var proj = new XamarinAndroidApplicationProject () {
-				IsRelease = true,
-				AotAssemblies = true,
+				IsRelease = isRelease,
+				AotAssemblies = aotAssemblies,
```

Comment:
Maybe we should add a helper method for AotAssemblies, where you pass in the AndroidRuntime? And it returns true for Mono and false for the others.

Reply:
I was considering doing that, yeah, but then I decided that tests should be explicit as much as possible in what they do (even if it's repetitive/verbose), they should be treated as separate entities without hidden code that does something. 
I made an exception in the `IgnoreUnsupportedConfiguration` helper method because what it does can change in the future and it would be a real shame to have to go through all the tests and change them individually.
Another exception to this, but justified I think, is the `SetRuntime` method when called for `NativeAOT`, that's just environment setup unrelated to any particular test as such, so I think it's fine.